.travis.yml

```diff
@@ -17,7 +17,7 @@ matrix:
 install:
   - pip install --upgrade tox coveralls
 
-script: tox
+script: tox --recreate test
```

Comment:
This is fine, but technically we are this close 👌 to not needing this file anymore!  travis is still running builds but those builds only go on to power the coverage reporting service.  Once we figure out how to use github actions for that, we can eliminate travis altogether.

Reply:
oh come to think of it, the current github actions are using the example build scripts that are provided by github, which, I believe, means that it ignores our linter configuration and potentially some other stuff?  I guess we could ask ourselves whether we want to continue with tox, in which case we probably should switch the github actions to use tox, or otherwise we could eliminate tox altogether?  But if we eliminate it, then it's more annoying to run the tests locally...  so maybe we should make the github actions use tox?

I had no reason not to use tox with them initially, other than that I was more concerned with getting the actions to run properly at all, than i was with the specifics of how we run the tests.

Reply:
oh, good point. I'd lean toward using tox for the github actions too.

Reply:
Switched the github action over to tox as part of this PR and it seems fine (amazingly).  Tox is set up to use `pylint` and the github action is using `flake8`.  We should drive the linting through tox as well so that it's easier for developers to lint their code without reference to the configuration in the github action.

@mchalek which tool would you prefer we standardize on? I don't really have a preference.